tests/e2e/util.go

```diff
@@ -1907,3 +1908,112 @@ func checkHostStatus(ip string) error {
 	})
 	return waitErr
 }
+
+// getK8sMasterIP gets k8s master ip in vanilla setup
+func getK8sMasterIP(ctx context.Context, client clientset.Interface) string {
+	var err error
+	nodes, err := client.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	var k8sMasterIP string
+	for _, node := range nodes.Items {
+		if strings.Contains(node.Name, "master") {
+			addrs := node.Status.Addresses
+			for _, addr := range addrs {
+				if addr.Type == v1.NodeExternalIP {
+					k8sMasterIP = addr.Address
+				}
+			}
+		}
+	}
+	return k8sMasterIP
+}
+
+// toggleCSIMigrationFeatureGatesOnKubeControllerManager adds/removes CSIMigration and CSIMigrationvSphere feature gates to/from kube-controller-manager
+func toggleCSIMigrationFeatureGatesOnKubeControllerManager(ctx context.Context, client clientset.Interface, add bool) error {
+	var err error
+	sshCmd := ""
+	if !vanillaCluster {
+		return fmt.Errorf("'toggleCSIMigrationFeatureGatesToKubeControllerManager' is implemented for vanilla cluster alone")
+	}
+	if add {
+		sshCmd = "sed -i -e 's/CSIMigration=false,CSIMigrationvSphere=false/CSIMigration=true,CSIMigrationvSphere=true/g' " + kcmManifest
+	} else {
+		sshCmd = "sed -i '/CSIMigration/d' " + kcmManifest
+	}
+	grepCmd := "grep CSIMigration " + kcmManifest
+	k8sMasterIP := getK8sMasterIP(ctx, client)
+	framework.Logf("Invoking command %v on host %v", grepCmd, k8sMasterIP)
+	sshClientConfig := &ssh.ClientConfig{
+		User: "root",
+		Auth: []ssh.AuthMethod{
+			ssh.Password("ca$hc0w"),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	result, err := sshExec(sshClientConfig, k8sMasterIP, grepCmd)
+	if err != nil {
+		return err
+	}
+	if err != nil {
+		fssh.LogResult(result)
+		return fmt.Errorf("command failed/couldn't execute command: %s on host: %v , error: %s", grepCmd, k8sMasterIP, err)
+	}
+	if result.Code != 0 {
+		if add {
+			sshCmd = "gawk -i inplace '/--bind-addres/ { print; print \"    - --feature-gates=CSIMigration=true,CSIMigrationvSphere=true\"; next }1' " + kcmManifest
+		} else {
+			return nil
+		}
+	}
+	framework.Logf("Invoking command %v on host %v", sshCmd, k8sMasterIP)
+	result, err = sshExec(sshClientConfig, k8sMasterIP, sshCmd)
+	if err != nil || result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s", sshCmd, k8sMasterIP, err)
+	}
+	framework.Logf("Waiting for 'kube-controller-manager' controller pod to come up within %v seconds", pollTimeout)
+	time.Sleep(10 * time.Second)
```

Comment:
Add more time to be safe? If possible we can also check Pod Status to make sure Pod is up and running after specified wait interval.

Reply:
moved this sleep before the log message, it was added to wait for the change in manifest file to be recognised, the following code `WaitForPodsWithLabelRunningReady` actually checks for 'kube-controller-manager' pod to be ready and running and has a timeout of 5mins